Makefile

```diff
@@ -1,2 +1,110 @@
+endif
+
+_update_yamls: _set_registry_url
+	sed -i "s|plugin.registry.url: .*|plugin.registry.url: http://$(PLUGIN_REGISTRY_HOST)|g" ./deploy/controller_config.yaml
+	sed -i "s|image: .*|image: $(IMG)|g" ./deploy/controller.yaml
+	sed -i "s|imagePullPolicy: Always|imagePullPolicy: $(PULL_POLICY)|g" ./deploy/controller.yaml
+	sed -i "s|che.webhooks.enabled: .*|che.webhooks.enabled: $(WEBHOOK_ENABLED)|g" ./deploy/controller.yaml
+
+_reset_yamls: _set_registry_url
+	sed -i "s|http://$(PLUGIN_REGISTRY_HOST)|http://che-plugin-registry.192.168.99.100.nip.io/v3|g" ./deploy/controller_config.yaml
+	sed -i "s|image: $(IMG)|image: quay.io/che-incubator/che-workspace-controller:nightly|g" ./deploy/controller.yaml
+	sed -i "s|imagePullPolicy: $(PULL_POLICY)|imagePullPolicy: Always|g" ./deploy/controller.yaml
+	sed -i "s|che.webhooks.enabled: .*|che.webhooks.enabled: "false"|g" ./deploy/controller.yaml
+
+_update_crds:
+	$(TOOL) apply -f ./deploy/crds
+	$(TOOL) apply -f ./deploy/controller_config.yaml
+
+_deploy_controller:
+	$(TOOL) apply -f ./deploy
+
+### docker: build and push docker image
+docker:
+	docker build -t $(IMG) -f ./build/Dockerfile .
+	docker push $(IMG)
+
+### webhook: generate certificates for webhooks and deploy to cluster
+webhook:
+ifeq ($(WEBHOOK_ENABLED),true)
+	./deploy/webhook-server-certs/deploy-webhook-server-certs.sh oc
+else
+	echo "Webhooks disabled, skipping certificate generation"
+endif
+
+### deploy: deploy controller to cluster
+deploy: _set_context _deploy_registry _update_yamls _update_crds webhook _deploy_controller _reset_yamls
+
+### restart: restart cluster controller deployment
+restart:
+ifeq ($(TOOL),oc)
+	oc patch deployment/che-workspace-controller \
+		-n che-workspace-controller \
+		--patch "{\"spec\":{\"template\":{\"metadata\":{\"annotations\":{\"kubectl.kubernetes.io/restartedAt\":\"$$(date --iso-8601=seconds)\"}}}}}"
+else
+	kubectl rollout restart -n $(NAMESPACE) che-workspace-controller
+endif
+
+### rollout: rebuild and push docker image and restart cluster deployment
+rollout: docker restart
+
+### update: update CRDs defined on cluster
+update: _update_yamls _update_crds _reset_yamls
+
+### uninstall: remove namespace and all CRDs from cluster
+uninstall:
+	$(TOOL) delete namespace $(NAMESPACE)
+	$(TOOL) delete customresourcedefinitions.apiextensions.k8s.io workspaceroutings.workspace.che.eclipse.org
+	$(TOOL) delete customresourcedefinitions.apiextensions.k8s.io workspaces.workspace.che.eclipse.org
+
+### local: set up cluster for local development
+local: _set_context _deploy_registry _set_registry_url _update_yamls _update_crds _reset_yamls
+
+### fmt: format all go files in repository
 fmt:
 	go fmt -x ./...
+
+.PHONY: help
+### help: print this message
+help: Makefile
+	@echo "Available rules:"
+	@sed -n 's/^### /    /p' $< | awk 'BEGIN { FS=":" } { printf "%-20s - %s\n", $$1, $$2 }'
```

Comment:
```suggestion
	@sed -n 's/^### /    /p' $< | awk 'BEGIN { FS=":" } { printf "%-20s -%s\n", $$1, $$2 }'
```
to fix 
![Screenshot_20200401_172341](https://user-images.githubusercontent.com/5887312/78148541-a0edfd80-743d-11ea-9bf1-f22ba78d4671.png)

Reply:
Good catch, thanks.